src/main/kotlin/org/jitsi/jicofo/conference/source/ValidatingConferenceSourceMap.kt

```diff
@@ -18,67 +18,302 @@
 package org.jitsi.jicofo.conference.source
 
 import org.jitsi.jicofo.ConferenceConfig
-import org.jitsi.jicofo.InvalidSSRCsException
-import org.jitsi.jicofo.SSRCValidator
-import org.jitsi.protocol.xmpp.util.MediaSourceGroupMap
-import org.jitsi.protocol.xmpp.util.MediaSourceMap
 import org.jitsi.utils.logging2.createLogger
 import org.jxmpp.jid.Jid
+import java.lang.IllegalStateException
 
-class ValidatingConferenceSourceMap : ConferenceSourceMap() {
+/**
+ * An extension of [ConferenceSourceMap] with an API for addition ([tryToAdd]) and removal ([tryToRemove]) of sources
+ * with validation. Note that the standard [add] and [remove] APIs are still available and do NOT perform validation.
+ *
+ * Validation means that the addition of sources does not lead to a conflict of SSRC or MSID with another endpoint
+ * in the conference, and that the addition/removal operations leave the endpoint's resulting set of sources valid.
+ *
+ * A set of sources is valid if there are no conflicts in SSRC or MSID, all sources within a group have the same MSID,
+ * all SSRCs within a group have a corresponding source, and other similar checks pass. See [validateEndpointSourceSet].
+ */
+class ValidatingConferenceSourceMap(
+    private val maxSsrcsPerUser: Int = ConferenceConfig.config.maxSsrcsPerUser
+) : ConferenceSourceMap() {
     val logger = createLogger()
 
+    /**
+     * Maps an SSRC to the JID of the endpoint that owns it. Used to detect cross-endpoint conflicts efficiently.
+     *
+     * Note that since the owner is nullable [Map.containsKey] must be used when checking for.
```

Comment:
"when checking for" is missing an object, presumably "map membership" or the like.  Likewise below.